src/main/java/org/motechproject/mots/management/LocationImporter.java

```diff
@@ -0,0 +1,206 @@
+  private static final String FACILITY_HEADER = "Health Facility";
+  private static final String COMMUNITY_HEADER = "Community";
+
+
+  private List<District> districtList;
+  private List<Chiefdom> chiefdomList;
+  private List<Facility> facilityList;
+  private List<Community> communityList;
+
+  /**
+   * Initializes locationService and locations lists.
+   * @param locationService service to autowire.
+   */
+  @Autowired
+  public LocationImporter(LocationService locationService) {
+    this.locationService = locationService;
+    if (loadLocations) {
+      this.districtList = locationService.getDistricts();
+      this.chiefdomList = locationService.getChiefdoms();
+      this.facilityList = locationService.getFacilites();
+      this.communityList = locationService.getCommunities();
+    }
+  }
+
+  /**
+   * Implementation of run method. A container for all actions which loads locations data to db.
+   * @param args Arguments passed to the application
+   * @throws IOException if location of the .xlsx file is wrong.
+   */
+  public void run(ApplicationArguments args) throws IOException {
+    if (!loadLocations) {
+      return;
+    }
+
+    InputStream excelFileToRead = new FileInputStream("src/main/resources/SL_locations.xlsx");
+    XSSFWorkbook  wb = new XSSFWorkbook(excelFileToRead);
+
+    XSSFSheet sheet = wb.getSheet(DISTRICT_HEADER);
+    parseDistricts(sheet);
+    this.districtList = locationService.getDistricts();
+    sheet = wb.getSheet(CHIEFDOM_HEADER);
+    parseChiefdoms(sheet);
+    this.chiefdomList = locationService.getChiefdoms();
+    sheet = wb.getSheet(FACILITY_HEADER);
+    parseFacilities(sheet);
+    this.facilityList = locationService.getFacilites();
+    sheet = wb.getSheet(COMMUNITY_HEADER);
+    parseCommunities(sheet);
+
+    LOGGER.info("Locations have been successfully loaded");
+  }
+
+  private void parseDistricts(XSSFSheet sheet) {
+    XSSFRow row;
+    XSSFCell cell;
+    Iterator rows = sheet.rowIterator();
+
+    while (rows.hasNext()) {
+      row = (XSSFRow) rows.next();
+      cell = row.getCell(0);
+      String cellText = cell.getStringCellValue();
+
+      if (cellText.equals(DISTRICT_HEADER)) {
+        continue;
+      }
+
+      District district = new District(cellText);
+
+      if (!districtList.contains(district)) {
+        locationService.createDistrict(new District(cellText));
+      }
+    }
+  }
+
+  private void parseChiefdoms(XSSFSheet sheet) {
+    XSSFRow row;
+    XSSFCell cell;
+    Iterator rows = sheet.rowIterator();
+
+    while (rows.hasNext()) {
+      row = (XSSFRow) rows.next();
+      cell = row.getCell(0);
+      String cellText = cell.getStringCellValue();
+
+      if (cellText.equals(CHIEFDOM_HEADER)) {
+        continue;
+      }
+
+      Chiefdom chiefdom = new Chiefdom(cellText);
+      String parentName = row.getCell(1).getStringCellValue();
+
+      Optional<District> parent = districtList.stream()
```

Comment:
use District instead of Optional<District> and at the end add .orElseThrow(() -> new RuntimeException())
I'm not sure what type of exception will be best here maybe create some custom exception for this